WcaOnRails/app/models/competition.rb

```diff
@@ -1681,6 +1682,22 @@ def set_wcif!(wcif, current_user)
     end
   end
 
+  def set_wcif_competitor_limit!(wcif_competitor_limit, current_user)
+    if confirmed? && !current_user.can_admin_competitions?
```

Comment:
`can_admin_competition` is the right permission here :)
Using something like `can_manage_competition?(self)` will include competition Delegates and their Senior delegates, whereas the former only includes global results admins (WST and WRT) as well as WCAT